x/twap/logic.go

```diff
@@ -190,6 +192,12 @@ func recordWithUpdatedAccumulators(record types.TwapRecord, newTime time.Time) t
 	p1NewAccum := types.SpotPriceMulDuration(record.P1LastSpotPrice, timeDelta)
 	newRecord.P1ArithmeticTwapAccumulator = newRecord.P1ArithmeticTwapAccumulator.Add(p1NewAccum)
 
+	// logP0SpotPrice = log_{1.0001}{P_0}
+	logP0SpotPrice := osmomath.TwapLog(record.P0LastSpotPrice)
+	// p0NewGeomAccum = log_{1.0001}{P_0} * timeDelta
+	p0NewGeomAccum := types.SpotPriceMulDuration(logP0SpotPrice, timeDelta)
+	newRecord.GeometricTwapAccumulator = newRecord.GeometricTwapAccumulator.Add(p0NewGeomAccum)
```

Comment:
Why is it enough to store this for P0? Is it cause we can later obtain p1 via `1/p0`?

Reply:
Exactly. We use this property: Geometric mean of the reciprocals is the reciprocal of the geometric mean.

I attached the proof in the computeGeometricTwap function.